homeassistant/components/private_ble_device/__init__.py

```diff
@@ -2,15 +2,44 @@
 
 from __future__ import annotations
 
-from homeassistant.config_entries import ConfigEntry
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_INTEGRATION_DISCOVERY, ConfigEntry
 from homeassistant.const import Platform
-from homeassistant.core import HomeAssistant
+from homeassistant.core import HomeAssistant, ServiceCall, SupportsResponse
+from homeassistant.helpers import config_validation as cv, discovery_flow
+
+from .const import DOMAIN
 
 PLATFORMS = [Platform.DEVICE_TRACKER, Platform.SENSOR]
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up tracking of a private bluetooth device from a config entry."""
+
+    async def service_register_irk(call: ServiceCall) -> None:
+        """Trigger Private BLE Device to add an IRK."""
+        if _irk := call.data.get("irk", False):
+            discovery_flow.async_create_flow(
+                hass,
+                DOMAIN,
+                context={"source": SOURCE_INTEGRATION_DISCOVERY},
+                data={"irk": _irk},
+            )
+
+    # Register action for creating new Private_BLE_Devices
+    hass.services.async_register(
+        DOMAIN,
+        "register_irk",
+        service_register_irk,
+        vol.Schema(
+            {
+                vol.Required("irk"): cv.string,
+            }
+        ),
+        supports_response=SupportsResponse.NONE,
+    )
```

Comment:
I'm concerned about the security implications of allowing a service call to create a config entry. At the very least, this should be restricted to admin-level access.

That said, it might be better to expose this as a top-level API on the integration instead. That way, it's clearly scoped for use by other integrations and not exposed as a general-purpose service.